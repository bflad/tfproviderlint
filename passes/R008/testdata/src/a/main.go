package a

import (
	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
)

func f() {
	var d schema.ResourceData

	// Comment ignored

	//lintignore:R008
	d.SetPartial("test")

	d.SetPartial("test") //lintignore:R008

	// Failing

	d.SetPartial("test") // want "deprecated \\(schema.ResourceData\\).SetPartial"

	fResourceFunc(&d, nil)
}

func fResourceFunc(d *schema.ResourceData, meta interface{}) error {
	d.SetPartial("test") // want "deprecated \\(schema.ResourceData\\).SetPartial"

	return nil
}
