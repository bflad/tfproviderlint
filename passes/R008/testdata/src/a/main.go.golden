package a

import (
	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
)

func f() {
	var d schema.ResourceData

	// Comment ignored

	//lintignore:R008
	d.SetPartial("test")

	d.SetPartial("test") //lintignore:R008

	// Failing

	// want "deprecated d.SetPartial"

	fResourceFunc(&d, nil)
}

func fResourceFunc(d *schema.ResourceData, meta interface{}) error {
	// want "deprecated d.SetPartial"

	return nil
}
